Repository/ServiceEntityRepository.php

```diff
@@ -7,10 +7,11 @@
 use LogicException;
 use Symfony\Component\VarExporter\LazyGhostTrait;
 
+use function property_exists;
 use function sprintf;
 use function trait_exists;
 
-if (trait_exists(LazyGhostTrait::class)) {
+if (trait_exists(LazyGhostTrait::class) && property_exists(EntityRepository::class, '_entityName')) {
```

Comment:
This basically opts us out of the new `LazyServiceEntityRepository` for ORM 3. I need a little help to get this to work, @nicolas-grekas.

Reply:
Ideally, we would get rid of LazyServiceEntityRepository: I introduced it to keep BC with existing protected properties on EntityRepository. But since those are private in 3.0, we should now be able to deal with lazyness by decorating public/protected method. As a reminder, the purpose of the lazy proxy is to move the ` $manager = $registry->getManagerForClass($entityClass);` call out of the construction step.

Reply:
This PR still support ORM 2.x though.

Reply:
I've pushed a version with a proxy for `EntityRepository`.